rfcs/assert_precondition.md

```diff
@@ -45,11 +45,10 @@ with a `PRECONDITION_FAILED` result when `condition` is not truthy.
         .then(
           function(result) { ... },
           function(err) {
-            if (isUnsupported(err)) { // "Unsupported" case determined ad-hoc
-              assert_precondition(false, algorithm + ' not implemented');
-            } else {
-              assert_unreached("Threw an unexpected error: " + err.toString());
-            }
+            var supported = !isUnsupported(err); // "Unsupported" case determined ad-hoc
+            assert_precondition(supported, algorithm + ' not implemented');
```

Comment:
I realize there's an issue with descriptions that often comes up. Most of the assert methods generate failure message that assume the description describes the actual value or what *did* happen. But it's somewhat common for people to write descriptions saying what the value shouldn't be or what shouldn't happen. That leads to pretty bad failure messages.

How will the failure message be constructed for this? The prefix "Precondition not met: " would require a positive description of what is (should) be supported, but is that the format you had in mind?

Since it's so easy to get this wrong, a clear example would be great. It'll still be wrong 30% of the time though :)